### PR TITLE
docs: DecodeSliceOfJSON in README.md

### PR DESCRIPTION
**Pull request by sgasho**
related: #521

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -502,4 +582,4 @@ go generate
 ### Testing
 
 Please use the [./dockertest.sh](./dockertest.sh) script for running test cases locally.
-And please try your best to have 100% test coverage on code changes.
\ No newline at end of file
+And please try your best to have 100% test coverage on code changes.
```
Comment (sgasho) [inline README.md:585]: I have no idea the differences....

File: README.md [hunk 2]
```diff
@@ -484,6 +484,86 @@ client.Do(ctx, client.B().FtSearch().Index("idx").Query("@f:v").Build()).AsFtSea
...
+	return err
+}
+
+
+users := make([]*User, 0)
+if err := rueidis.DecodeSliceOfJSON(client.Do(ctx, client.B().Mget().Key("user1").Build()), &users); err != nil {
+	return err
+}
+// users = [{name:name1}]
+```
```
Comment (sgasho) [inline README.md:549]: I wonder if this section is necessary...
for some experienced golang developer don't need it but some people like beginners or not familiar with golang might do.

File: README.md [hunk 3]
```diff
@@ -484,6 +484,86 @@ client.Do(ctx, client.B().FtSearch().Index("idx").Query("@f:v").Build()).AsFtSea
...
+
+DecodeSliceOfJSON is useful when you would like to scan the results of an array into a slice of a specific struct.
+
+```golang
+type User struct {
+	Name string
+}
+
+// Set some values
+if err = client.Do(ctx, client.B().Set().Key("user1").Value(`{"Name": "name1"}`).Build()).Error(); err != nil {
```
Comment (rueian) [inline README.md:497]:
```suggestion
if err = client.Do(ctx, client.B().Set().Key("user1").Value(`{"name": "name1"}`).Build()).Error(); err != nil {
```

File: README.md [hunk 4]
```diff
@@ -484,6 +484,86 @@ client.Do(ctx, client.B().FtSearch().Index("idx").Query("@f:v").Build()).AsFtSea
...
+	return err
+}
+if err = client.Do(ctx, client.B().Set().Key("user2").Value(`{"Name": "name2"}`).Build()).Error(); err != nil {
```
Comment (rueian) [inline README.md:500]:
```suggestion
if err = client.Do(ctx, client.B().Set().Key("user2").Value(`{"name": "name2"}`).Build()).Error(); err != nil {
```

File: README.md (hunk 2 above)
Comment (rueian) [reply README.md:549]: Indeed, I think this section is not necessary, and scanning into private fields is also not a common approach. Actually using `json` tags is preferred in the above case.

File: README.md [hunk 5]
```diff
@@ -484,6 +484,54 @@ client.Do(ctx, client.B().FtSearch().Index("idx").Query("@f:v").Build()).AsFtSea
...
+	return err
+}
+
+users := make([]*User, 0)
+if err := rueidis.DecodeSliceOfJSON(client.Do(ctx, client.B().Mget().Key("user1").Build()), &users); err != nil {
+	return err
+}
+// -> Error: invalid character 'u' looking for beginning of value
+// in this case, use client.Do(ctx, client.B().Mget().Key("user1").Build()).AsStrSlice()
+```
```
Comment (rueian) [inline README.md:533]: Hi @sgasho,

I understand that you want to point out an incorrect usage, but I am afraid that the warning is not obvious enough.

Users would have to look at this example very carefully, including the final comments, to know that this is a counterexample. Could you try to add more warnings around the code? For example, adding something like `!!!!!! DO NOT DO THIS !!!!!!`

**Review by rueian — APPROVED** [review]
Thanks @sgasho!

**Pull request merged**